### ISSUE TITLE
Change request: Improving change management

### ISSUE THREAD

**Issue by jamalv** [teamforus/general#31]
## Change request

Requested by: stakeholders

### Change description

As a stakeholder I would like to know what the status of a change is, so that I know what to expect. I would also like an easy way to request a change.

## Change livecycle.
Every change goes trough this lifecycle to ensure the highest quality result.

Phase 1: Analysis
Phase 2: Proposal
Phase 3: Prototyping
Phase 4: Development
Phase 5: Staging
Phase 6: Testing
Phase 7: Release

## Phase 1: Analysis
Goal: Research if this change is desirable/feasable & what the impact would be.

Type of change:

This change would impact:
- [ ] Design
- [ ] Code
- [ ] Infrastructure
- [ ] Users
- [ ] Finances
- [x] Processes

Analysis tasks:
- [ ] Research feasibility
- [ ] Perform risk assesment
- [ ] Need to inform users

Make decision
- [ ] Go
- [ ] No Go

## Phase 2: Proposal
Goal: Make a concrete proposal for the change.

- [ ] Specifications
- [ ] Communciation plan
- [ ] Planning in time

Make decision
- [ ] Go
- [ ] No Go

## Phase 3: Prototyping
Goal: Create high fidelity prototype for approval and developer handoff.

Figma link:

- [ ] Add figma link
- [ ] Create change in figma
- [ ] Add content

Make decision
- [ ] Go
- [ ] No Go

## Phase 4: Development
Goal: Convert the prototype into code and add changes into the design library.

(draft) Pull request link: 

- [ ] Create a new feature branch and open draft pull request
- [ ] Create pull request file in figma and prepare new versions for library
- [ ] Review pull requests

Make decision
- [ ] Go
- [ ] No Go

## Phase 5: Staging
Goal: Add the new feature to an upcoming release and merge the changes into the design library

Release pull request link: 

- [ ] Merge into release branch and add to staging environment

## Phase 6: Testing
Goal: Ensure that the new feature is stable and does not break anything

- [ ] Run tests on staging environment
- [ ] Fix bugs if needed
- [ ] Revert merge if bugs not fixable for this release

## Phase 7: Release
Goal: Add the feature into production and inform users as planned in phase 2

- [ ] Inform users of (upcoming) change
- [ ] Release feature on a monday morning

**Comment by maxvisser** [comment]
My proposal:
RACI for each phase

R = Responsible (also Recommender)
Those who do the work to complete the task.[ There is at least one role with a participation type of responsible, although others can be delegated to assist in the work required (see also RASCI below for separately identifying those who participate in a supporting role).
A = Accountable (also Approver or final approving authority)
The one ultimately answerable for the correct and thorough completion of the deliverable or task, the one who ensures the prerequisites of the task are met and who delegates the work to those responsible. In other words, an accountable must sign off (approve) work that responsible provides. There must be only one accountable specified for each task or deliverable.
C = Consulted (sometimes Consultant or counsel)
Those whose opinions are sought, typically subject matter experts; and with whom there is two-way communication.
I = Informed (also Informee)
Those who are kept up-to-date on progress, often only on completion of the task or deliverable; and with whom there is just one-way communication.

Example setup
Phase 1
R: Jamal (works on analysing problem)
A: Max (checks if jamal has analysed the problem)
C: Gerben (gives feedback on jamal's analysis), refinement session whole team
I: clients, maarten, linda

For each feature request I would start with dividing this within team members, who is responsible for delivering the first phase, who will check if its done and who can be consulted for the analyzation


Another example
Phase 7
R: Max releases on monday morning
A: Gerben check if users are informed and if release is indeed online
C: 
I: Jamal informs users

**Comment by maxvisser** [comment]
for each phase I would like a Definition of Done.

**Comment by maxvisser** [comment]
Analyses result in a Improvement proposal
Improvement proposal results in a prototype
Prototype results in a feature branch merged to develop
Develop will be staged as a release with multiple features;

I would say after staging this process ends; as 6 and 7 are always done in a bundle

**Comment by maxvisser** [comment]
After writing this down; I think Analysis should be after IP; as you can't assess risks or feasibility if you don't know what the proposal is about.

**Comment by maxvisser** [comment]
made two templates one for improvement proposal and one for analysis

**Comment by maxvisser** [comment]
I would also like to seperate Prototyping in three phases

## Phase 3: Wireframe (if needed)
<< brainstorm session results in a photo of a whiteboard, picture of a drawing on paper >>

<< R ask C for feedback or to join brainstorm session; R asks A to approve wireframe; R informs I of wireframe if done; >>

## Phase 4: Mockup / Template
<< create a fast mockup template with current library >>
<< design new elements if needed >>
<< ask designer to look at new template >>

<< R ask C for feedback; R asks A to approve wireframe; R informs I of wireframe if done; >>

## Prototype
<< Use templates to create a prototype (definition: high fidelity representation of the final product which is meant to simulate user interaction) in figma >>

<< R ask C for feedback on prototype; R asks A to approve wireframe; R informs I of wireframe if done; >>

**Comment by maxvisser** [comment]
braindump; How I see what the workflow could be:

-- how your request gets added to forus its backlog --
User starts Improvement proposal fills in improvement proposal (should be easy and understandable, low effort)

Product owner analysis improvement proposal gives go/no go together with consulted team members if needed
--

teamforus decides to work on your proposal; could be weeks or months after improvement proposal

-- how ui/ux gets decided of your improvement proposal --
Design team sprintplanning makes wireframe issue for IP number. Initial creator gets informed of starting on its IP
Design team decides who will wireframe improvement proposal; who will help if assignee gets stuck if needed and development team will accept wireframe. if IP got unexpectedly bigger or changed. IP should first updated or questions about IP need to answered. Scope of wireframe should never be bigger then initial IP; it may only be smaller.

If development team accepted Wireframe; IP can be designed where template / mockups can be made.

Templates will be made to a prototype which can be shared by users. Feedback should always be added to IP; giving users prototype can lead to new improvement proposals; which can be wireframed and made into templates updates; 

prototype is done when a user has held the prototype in their hands and used it to perform the user story which was described initially in the IP. 

prerequites: API credentials, infrastructure update which are needed for development should be investigated after wireframe phase if needed
--

Prototype is ready to be build by developers and added a workflow change or new functionality to the platform.

-- process is more or less clear already; IP with a finished prototype is planned for sprint --

**Comment by jamalv** [comment]
Statement: We should host our change requests (CR's) in a public repository so that is easy to communicate to all our stakeholders what the status of a chance is.

When we need to have a discussion about a specific topic that should probably not be public (e.g. security or infrastructure) we will open a issue in an internal repository that links to the CR.

Thumbs up if you agree, thumbs down if you disagree.

**Comment by maartenfv** [comment]
What about a last phase of evaluation? After something has been released it would be nice to gather some feedback to see how it actually landed in the real world. By adding an additional label you could track this.

**Comment by maartenfv** [comment]
Who is responsible for labels? Needs to be one person I think.

**Comment by TomForus** [comment]
**Priority labels**
I think at the end 1 person should be responsible for maintaining the labels. This person should have an overview of all the issues.

- Might add a **Priority: Needs dicussion** label
Because it could happen that due to a discussion the type of priority could change.

My proposal will be: Jamal is responsible for the priority labels. He could ask someone else to add the label (for example Gerben (Customer support) or Max (infra) but at the end there should be one person responsible

**Comment by maartenfv** [comment]
Please add a source label

- Aanbesteding
- Helpdesk
- Development team
- Customer request
- etc..

**Comment by jamalv** [comment]
@maxvisser I see that you updated the template. Could you restore to each phase: 
- [ ] at least one checkbox that the phase / steps in the phase have been completed. 

A 
- [ ] go / 
- [ ] no go 

decision?

This way we ensure that everything does actually happen, and that a decision to proceed has been made.

**Comment by maxvisser** [comment]
added tbe checkboxes; personally I don't like the big templates and don't see the need for so many go/no go moments;

**Comment by jamalv** [comment]
We have created manuals for the different types of issues that are involved with change management: https://github.com/teamforus/general/blob/develop/manuals/development/issues.md

We have also updated the issue templates and labels to match this process.

What's still left to do is to add the responsibilities part, I think defining roles and responsibilities is out of scope for this change request, though. Therefore I am closing this issue.
